aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -101,4 +104,199 @@ Tensor& randperm_out_cuda(Tensor& result, int64_t n, Generator* generator) {
   return result;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ triangle ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+namespace {
+// f: the number of elements in the first row of the trapezoid.
+// x: the index of the target coordinates ordered by row and then column.
```

Comment:
x is an int64_t -- did you pack (row, column) in as two int32_t or is `x` just a linear index?

Reply:
nvm I see, it is a linear index